aggregator/election_mgr.go

```diff
@@ -320,18 +400,23 @@ func (mgr *electionManager) Resign(ctx context.Context) error {
 
 func (mgr *electionManager) Close() error {
 	mgr.Lock()
-	defer mgr.Unlock()
-
 	if mgr.state != electionManagerOpen {
+		mgr.Unlock()
 		return errElectionManagerNotOpenOrClosed
 	}
 	close(mgr.doneCh)
 	mgr.state = electionManagerClosed
+	mgr.Unlock()
+
```

Comment:
close the watchables so whoever is listening on them can receive the final signals?